aten/src/ATen/core/boxing/impl/boxing.h

```diff
@@ -7,73 +7,252 @@
+// 1. Unsupported type traps.
+//
+// These specializations capture the remaining gaps in boxing support.
+// Rather than triggering compile errors, we generate boxed kernels that
+// raise runtime errors. As support for these types is added, the
+// specializations can be removed.
+//
 
-template <class Result, class... Args>
-using supports_boxing =
-  guts::negation<guts::disjunction<
-    std::is_lvalue_reference<Result>,
-    not_ok_to_box<Result>,
-    std::is_same<IntArrayRef, Result>,
-    not_ok_to_box<std::decay_t<Args>>...
-  >>;
+// at::Dimname
+template <class... Args>
+using has_dimname_arg =
+  guts::disjunction<
+    std::is_same<at::Dimname, std::decay_t<Args>>...,
+    std::is_same<c10::ArrayRef<at::Dimname>, std::decay_t<Args>>...,
+    std::is_same<c10::optional<c10::ArrayRef<at::Dimname>>, std::decay_t<Args>>...
+  >;
 
 template<class Result, class... Args>
-Result boxAndCallBoxedFunc(KernelFunction::InternalBoxedKernelFunction* boxed_kernel_func, OperatorKernel* functor, const OperatorHandle& opHandle, Args... args, std::enable_if_t<!supports_boxing<Result, Args...>::value, int> = 0) {
-  // Some kernels don't need to actually box, and don't return.  If that's the
-  // case, just call them anyway without a stack.  These special cases can be
-  // removed once we support boxing everything.
-  // See Note [named_not_supported_kernel]
-  if (boxed_kernel_func == &named_not_supported_kernel) {
-    named_not_supported_kernel(functor, opHandle, nullptr);  // does not return
+struct BoxedKernelWrapper<Result(Args...), std::enable_if_t<has_dimname_arg<Args...>::value, void>> {
+  static Result call(KernelFunction::InternalBoxedKernelFunction*, OperatorKernel*, const OperatorHandle&, Args... args) {
+    TORCH_INTERNAL_ASSERT(false, "Call to a boxed kernel with unboxable parameter type at::Dimname.");
   }
+};
 
-  TORCH_INTERNAL_ASSERT(false, "Tried to call KernelFunction::call() for a kernel that only has a boxed kernel and doesn't support calling from an unboxed API yet.");
-}
+// at::Quantizer
+template <class... Args>
+using has_quantizer_arg =
+  guts::disjunction<
+    std::is_same<at::Quantizer, std::decay_t<Args>>...,
+    std::is_same<c10::intrusive_ptr<at::Quantizer>, std::decay_t<Args>>...
+  >;
 
-// SFINAE version for ops with returns
 template<class Result, class... Args>
-std::enable_if_t<supports_boxing<Result, Args...>::value && !std::is_same<void, Result>::value, Result>
-boxAndCallBoxedFunc(KernelFunction::InternalBoxedKernelFunction* boxed_kernel_func, OperatorKernel* functor, const OperatorHandle& opHandle, Args... args) {
-  // TODO Reuse stack vector instead of allocating?
-  torch::jit::Stack stack;
-  torch::jit::push(stack, std::forward<Args>(args)...);
+struct BoxedKernelWrapper<Result(Args...), std::enable_if_t<has_quantizer_arg<Args...>::value, void>> {
+  static Result call(KernelFunction::InternalBoxedKernelFunction*, OperatorKernel*, const OperatorHandle&, Args... args) {
```

Comment:
Have you tried a version where those unsupported cases also use `if_constexpr` instead of SFINAE? Is it more readable?

Reply:
tried it but, but it wasn't more readable. also, these should go away as support is added